core/src/latest_unprocessed_votes.rs

```diff
@@ -0,0 +1,675 @@
+use {
+    crate::{
+        banking_stage::weighted_random_order_by_stake,
+        forward_packet_batches_by_accounts::ForwardPacketBatchesByAccounts,
+        immutable_deserialized_packet::{DeserializedPacketError, ImmutableDeserializedPacket},
+    },
+    solana_perf::packet::{Packet, PacketBatch},
+    solana_sdk::{clock::Slot, program_utils::limited_deserialize, pubkey::Pubkey},
+    solana_vote_program::vote_instruction::VoteInstruction,
+    std::{
+        cell::RefCell,
+        collections::HashMap,
+        ops::DerefMut,
+        rc::Rc,
+        sync::{
+            atomic::{AtomicUsize, Ordering},
+            RwLock,
+        },
+    },
+};
+
+#[derive(PartialEq, Eq, Debug, Copy, Clone)]
+pub enum VoteSource {
+    Gossip,
+    Tpu,
+}
+
+/// Holds deserialized vote messages as well as their source, foward status and slot
+#[derive(Debug, Clone)]
+pub struct DeserializedVotePacket {
+    vote_source: VoteSource,
+    pubkey: Pubkey,
+    vote: Option<Rc<ImmutableDeserializedPacket>>,
+    slot: Slot,
+    forwarded: bool,
+}
+
+impl DeserializedVotePacket {
+    pub fn new(packet: Packet, vote_source: VoteSource) -> Result<Self, DeserializedPacketError> {
+        if !packet.meta.is_simple_vote_tx() {
+            return Err(DeserializedPacketError::VoteTransactionError);
+        }
+
+        let vote = ImmutableDeserializedPacket::new(packet, None)?;
+        let message = vote.transaction().get_message();
+        let (_, instruction) = message
+            .program_instructions_iter()
+            .next()
+            .ok_or(DeserializedPacketError::VoteTransactionError)?;
+
+        match limited_deserialize::<VoteInstruction>(&instruction.data) {
+            Ok(VoteInstruction::UpdateVoteState(vote_state_update))
+            | Ok(VoteInstruction::UpdateVoteStateSwitch(vote_state_update, _)) => {
```

Comment:
I think ideally we could add a field to the packet meta that would carry the slot. This could also supersede the `is_simple_vote` flag. Seems like an involved change so i was planning on doing it separately - what do you think?

Reply:
https://github.com/solana-labs/solana/issues/27556